src/presence_analyzer/tests.py

```diff
@@ -71,20 +127,127 @@ def tearDown(self):
         """
         pass
 
-    def test_get_data(self):
+    @patch('presence_analyzer.utils.log')
+    def test_get_data(self, mock_log):
         """
         Test parsing of CSV file.
         """
         data = utils.get_data()
         self.assertIsInstance(data, dict)
-        self.assertItemsEqual(data.keys(), [10, 11])
+        self.assertItemsEqual(data.keys(), [10, 11, 12])
         sample_date = datetime.date(2013, 9, 10)
         self.assertIn(sample_date, data[10])
         self.assertItemsEqual(data[10][sample_date].keys(), ['start', 'end'])
         self.assertEqual(
             data[10][sample_date]['start'],
             datetime.time(9, 39, 5),
         )
+        self.assertTrue(mock_log.debug.called)
```

Comment:
I run coverage and noticed that the test does not cover these lines from utils.py line 59-61:
```
            if len(row) != 4:
                # ignore header and footer lines
                continue
```
It is very simple to add such a condition, it is enough to test less than 4 elements in the data file, example:
```
...
11,2013-09-13,13:16:56,15:04:02
11,2013/09/13,13:16:56,15:04:02
12,2013/09/14,13:16:56
```
I would like you to consider creating a separate file only with damaged data like: `broken_data.csv`. Then define this file in line 18: 
```
TEST_BROKEN_DATA_CSV = os.path.join(
    os.path.dirname(__file__), '..', '..', 'runtime', 'data', 'broken_data.csv'
)
```
And finally you can overwrite settings only for one test using this decorator:
```
@patch.dict('presence_analyzer.main.app.config', {'DATA_CSV': TEST_BROKEN_DATA_CSV})
```
if you create a separate file for damaged data, restore the original values in the file `test_data.csv`